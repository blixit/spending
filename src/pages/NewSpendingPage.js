import React from 'react';

import NewSpendingForm from 'components/forms/new-spending-form/NewSpendingForm';

import Page from 'pages/Page';

class NewSpendingPage extends React.Component {
  static displayName = 'NewSpendingPage';

  render () {
    return (
      <Page {...this.props}>
        <NewSpendingForm />
      </Page>
    );
  }
}

export default NewSpendingPage;
