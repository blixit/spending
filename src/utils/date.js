/**
 * Formats a date for the backend
 * @param {Date} date the date to convert
 */
export const toBackendDate = (date) => {
  return date
    ? date.toLocaleDateString() + ' ' + date.toLocaleTimeString()
    : date;
};