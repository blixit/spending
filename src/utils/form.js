/**
* Removes the listed fields from the source
* @param {*} form The source object
* @param {*} fields The fields list
*/
export const clean = (form, fields) => {
  const result = {};
  // eslint-disable-next-line no-unused-vars
  for (const field in form) {
    if (!form.hasOwnProperty(field) || fields.includes(field)) {
      continue;
    }
    result[field] = form[field];
  }
  return result;
};
