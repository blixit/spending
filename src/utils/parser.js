export const stringify = data => {
  try {
    return JSON.stringify(data);
  } catch (error) {
    return null;
  };
}

export const parse = data => JSON.parse(data);