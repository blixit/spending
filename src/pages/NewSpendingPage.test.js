import { withTheme } from 'ui/theming/test.utils';

import render from 'testing/render';

import NewSpendingPage from './NewSpendingPage';

describe('[pages] NewSpendingPage', () => {
  it('should render properly', () => {
    const tree = render(withTheme(NewSpendingPage, { children: 'Test'}));
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
