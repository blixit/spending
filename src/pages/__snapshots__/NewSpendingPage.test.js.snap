// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[pages] NewSpendingPage should render properly 1`] = `
<h1>
  Something went wrong.
</h1>
`;
